ethcore/src/state/mod.rs

```diff
@@ -108,6 +108,20 @@ impl AccountEntry {
 			state: AccountState::Clean,
 		}
 	}
+
+	// Replace data with another entry but preserve storage cache
+	fn merge_snapshot(&mut self, other: AccountEntry) {
+		self.state = other.state;
```

Comment:
Aren't the items in the cache dirtier than those in the snapshots? In `revert_snapshot` it merges the snapshotted account into the cached, overwriting the cached account with the (cleaner) snapshotted account while preserving the dirty storage (which `revert_snapshot` should be throwing away?)

The name is a bit confusing too, it seems like it would be used in `clear_snapshot` but is in fact only used when reverting.